### PR TITLE
Add maxHeadersCount and headersTimeout properties to HTTPS module in @types/node

### PR DESCRIPTION
**Pull request by tyronedougherty**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://nodejs.org/api/https.html#https_server_headerstimeout
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Review by SimonSchick — CHANGES_REQUESTED** [review]
Please add a quick test 😄

**Review by j-oliveras — CHANGES_REQUESTED** [review]
Can you add `maxHeadersCount` to node v10?

See https://nodejs.org/docs/latest-v10.x/api/https.html#https_server_maxheaderscount

File: types/node/https.d.ts [hunk 1]
```diff
@@ -26,7 +26,13 @@ declare module "https" {
 
         setTimeout(callback: () => void): this;
         setTimeout(msecs?: number, callback?: () => void): this;
+        maxHeadersCount: number;
```
[the remark this answers is not in the document]
Comment (Flarna) [reply types/node/https.d.ts]: Should be `number | null`, see https://github.com/nodejs/node/blob/v11.11.0/lib/https.js#L76

[the remark this answers is not in the document]
Comment (tyronedougherty) [reply types/node/https.d.ts]: Done :)

**Comment by tyronedougherty** [comment]
I've updated the PR with the requested changes :)

**Review by SimonSchick — APPROVED** [review]

**Comment by Flarna** [comment]
`headersTimeout` has been added in NodeJs 10.14 but v10/index.d.ts tells 10.12. Could you please update the first line in v10/index.d.ts to 10.14?

Not sure if we should really add links to source code as JsDoc here as this is usually not what you want to see during programming. But I will not block because of this.

**Review by BrunoScheufler — APPROVED** [review]

**Review by r3nya — APPROVED** [review]

**Review by j-oliveras — APPROVED** [review]

File: types/node/http.d.ts [hunk 2]
```diff
@@ -101,11 +101,13 @@ declare module "http" {
 
         setTimeout(msecs?: number, callback?: () => void): this;
         setTimeout(callback: () => void): this;
-        maxHeadersCount: number;
+        // https://nodejs.org/api/http.html#http_server_maxheaderscount
```
Comment (SimonSchick) [inline types/node/http.d.ts]: Uh, please use actual JSDoc eg. `/**` or just copy/paste the docs directly.

Comment (tyronedougherty) [reply types/node/http.d.ts]: I was just copying what I've seen throughout the typings... in fact none of the links in that file use JSDoc except the one that I added :/

**Comment by tyronedougherty** [comment]
> `headersTimeout` has been added in NodeJs 10.14 but v10/index.d.ts tells 10.12. Could you please update the first line in v10/index.d.ts to 10.14?

Done!

> Not sure if we should really add links to source code as JsDoc here as this is usually not what you want to see during programming. But I will not block because of this.

I see it throughout the typings here. I think it's ok :)

File: types/node/http.d.ts (hunk 2 above)
Comment (SimonSchick) [reply types/node/http.d.ts]: Yeea, docs are a little bit inconsistent but every bit helps!

**Review by Flarna — APPROVED** [review]

**Review by SimonSchick — APPROVED** [review]

**Pull request merged**